dcos/http/client/client.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2016 Mesosphere, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/dcos/dcos-go/jwt/transport"
+)
+
+// loadCAPool will load a valid x509 cert.
+func loadCAPool(path string) (*x509.CertPool, error) {
+	caPool := x509.NewCertPool()
+	f, err := os.Open(path)
```

Comment:
+1, appears os.Open, though used by ioutil.ReadFile, does not implement an implicit .Close() https://golang.org/src/io/ioutil/ioutil.go?s=1464:1510#L39